pecos/xmr/reranker/model.py

```diff
@@ -0,0 +1,601 @@
+        content_sep: str = " "
+        append_eos_token: bool = False
+        pad_to_multiple_of: Optional[int] = 8
+
+    @dataclass
+    class EvalParams(pecos.BaseParams):
+        """
+        Evaluation parameters
+        """
+
+        model_name_or_path: str
+        target_data_folder: str
+        input_data_folder: str
+        label_data_folder: str
+        output_dir: str
+        output_file_prefix: str = "output_"
+        output_file_suffix: str = ""
+        per_device_eval_batch_size: int = 128
+        dataloader_num_workers: int = 2
+        dataloader_prefetch_factor: int = 10
+        rerank_max_len: int = 196
+        query_prefix: str = "query: "
+        passage_prefix: str = "document: "
+        inp_id_col: str = "inp_id"
+        lbl_id_col: str = "lbl_id"
+        keyword_col_name: str = "keywords"
+        content_col_names: List[str] = field(default_factory=lambda: ["title", "contents"])
+        content_sep: str = " "
+        append_eos_token: bool = False
+        pad_to_multiple_of: int = 16
+        bf16: bool = True
+        device: str = "cuda"
+        model_init_kwargs: dict = dc.field(default_factory=lambda: dict())
+
+    def __init__(
+        self,
+        encoder: Union[CrossEncoder, PeftModel, PeftMixedModel],
+        tokenizer: AutoTokenizer,
+        model_params: ModelParams,
+        train_params: Optional[TrainParams] = None,
+    ):
+        """
+        Initialize the ranking model. The model contains the encoder, tokenizer, model parameters and training parameters.
+        """
+        self.tokenizer = tokenizer
+        self.cross_encoder = encoder
+
+        self.model_params = self.ModelParams.from_dict(model_params)
+        self.train_params = self.TrainParams.from_dict(train_params) if train_params else None
+
+    @classmethod
+    def get_modified_model(cls, model: CrossEncoder, mod_config: Dict):
+        """
+        Takes a pretrained Huggingface model and modifies it to include new features. Currently, the `modifier_type`
+        supported by this method is limited to the `peft` package.
+
+        Args:
+            model (PreTrainedModel): A PreTrainedModel from the transformers package.
+        """
+        if mod_config["modifier_type"] == "peft":
+            config_type = getattr(peft, mod_config["config_type"])
+            peft_config: PeftConfig = config_type(**mod_config["config"])
+
+            model = get_peft_model(model, peft_config)
+
+            return model
+        else:
+            logger.warn("Using model without modifiers (e.g. LoRA)")
+            return model
+
+    @classmethod
+    def init_model(cls, model_params: ModelParams, train_params: TrainParams):
+        """Initiate a model with training parameters
+
+        Args:
+            model_params (RankingModel.ModelParams): the model parameters
+            train_params (RankingModel.TrainParams): the training parameters
+        Returns:
+            An instance of UberGlobalModel
```

Comment:
Doc String seems to be out-dated. Remove `UberGlobalModel`.

Reply:
Updated docstrings